src/main/java/de/unistuttgart/overworldbackend/service/NPCService.java

```diff
@@ -92,6 +98,9 @@ public NPC getNPCFromDungeon(final int courseId, final int worldIndex, final int
    * @return the npc area as DTO
    */
   public NPCDTO updateNPCFromWorld(final int courseId, final int worldIndex, final int npcIndex, final NPCDTO npcDTO) {
+    List<PlayerNPCStatistic> statistics = playerNPCStatisticRepository.findByNpcId(npcDTO.getId());
+    List<PlayerNPCStatistic> courseStatistics = playerNPCStatisticRepository.findByCourseId(courseId);
+    statistics.forEach(statistic -> statistic.setCompleted(false));
```

Comment:
statistics does not get saved?

Reply:
Ah, I think I know why it doesn't:
That's a Hibernate "feature" if I remember that correctly: It uses `dynamic proxies` for objects it constructed meaning that any changes to the objects are immediately reflected in the database.
That's at least what I can remember, the last time I worked with Hibernate like that was about 4 years ago.